dbfit-java/informix/src/main/java/dbfit/environment/InformixEnvironment.java

```diff
@@ -0,0 +1,331 @@
+                   + "            WHEN   17 THEN 'INT8'"
+                   + "            WHEN   18 THEN 'SERIAL8'"
+                   + "            WHEN   19 THEN 'SET'"
+                   + "            WHEN   20 THEN 'MULTISET'"
+                   + "            WHEN   21 THEN 'LIST'"
+                   + "            WHEN   22 THEN 'ROW'"
+                   + "            WHEN   23 THEN 'COLLECTION'"
+                   + "            WHEN   24 THEN 'ROWDEF'"
+                   + "            WHEN   40 THEN 'LVARCHAR'"
+                   + "            WHEN   43 THEN 'LVARCHAR'"
+                   + "            WHEN   45 THEN 'BOOLEAN'"
+                   + "            WHEN   52 THEN 'BIGINT'"
+                   + "            WHEN   53 THEN 'BIGSERIAL'"
+                   + "            WHEN 2061 THEN 'IDSSECURITYLABEL'"
+                   + "            WHEN 4118 THEN 'ROW'"
+                   + "        END"
+                   + "           AS data_type"
+                   + "     , " + (forProcedureParameters ? "c.paramattr" : "1") // Dummy value for table/view columns.
+                   + "           AS direction"
+                   + "     , c." + (forProcedureParameters ? "paramid" : "colno")
+                   + "           AS position"
+                   + "  FROM informix." + (forProcedureParameters ? "sysprocedures p" : "systables p")
+                   + "     , informix." + (forProcedureParameters ? "sysproccolumns c" : "syscolumns c")
+                   + " WHERE p." + (forProcedureParameters ? "procid" : "tabid") + " = c." + (forProcedureParameters ? "procid" : "tabid")
+                   + "   AND ";
+
+        if (qualifiers.length == 2) {
+            qry += "LOWER(p.owner) = ? AND LOWER(p." + (forProcedureParameters ? "procname" : "tabname") + ") = ?";
+        } else {
+            qry += "LOWER(p." + (forProcedureParameters ? "procname" : "tabname") + ") = ?";
+        }
+
+        return qry;
+    }
+
+    public Map<String, DbParameterAccessor> getAllColumns(String tableOrViewName)
+            throws SQLException {
+        String[] qualifiers = NameNormaliser.normaliseName(tableOrViewName).split("\\.");
+        return readIntoParams(qualifiers, columnsOrParamtersQueryText(false, tableOrViewName));
+    }
+
+    private Map<String, DbParameterAccessor> readIntoParams(String[] queryParameters, String query) throws SQLException {
+        try (PreparedStatement dc = currentConnection.prepareStatement(query)) {
+            for (int i = 0; i < queryParameters.length; i++) {
+                dc.setString(i + 1, NameNormaliser.normaliseName(queryParameters[i]));
+            }
+
+            ResultSet rs = dc.executeQuery();
+            Map<String, DbParameterAccessor> allParams = new HashMap<String, DbParameterAccessor>();
+
+            while (rs.next()) {
+                String paramName = rs.getString(1);
+                if (paramName == null)
```

Comment:
In `SqlServerEnvironment` we're handling a similar expression using `defaultIfNull`. Might worth comparing to see if would be better to use the same approach here.